src/js/content_scripts/input-mapper.js

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright (c) 2020 The Gamepad Navigator Authors
+See the AUTHORS.md file at the top-level directory of this distribution and at
+https://github.com/fluid-lab/gamepad-navigator/raw/master/AUTHORS.md.
+
+Licensed under the BSD 3-Clause License. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the BSD 3-Clause License at
+https://github.com/fluid-lab/gamepad-navigator/blob/master/LICENSE
+*/
+
+/* eslint-env browser */
+
+(function (fluid) {
+    "use strict";
+
+    var gamepad = fluid.registerNamespace("gamepad");
+    fluid.registerNamespace("gamepad.inputMapperUtils");
+
+    fluid.defaults("gamepad.inputMapper", {
+        gradeNames: ["gamepad.navigator"],
+        model: {
+            actionMap: {
+                buttons: {
+                    "0": "vacant",
+                    "1": "vacant",
+                    "2": "vacant",
+                    "3": "vacant",
+                    "4": "vacant",
+                    "5": "vacant",
+                    "6": "vacant",
+                    "7": "vacant",
+                    "8": "vacant",
+                    "9": "vacant",
+                    "10": "vacant",
+                    "11": "vacant",
+                    "12": "scrollUp",
+                    "13": "scrollDown",
+                    "14": "scrollLeft",
+                    "15": "scrollRight",
+                    "16": "vacant"
+                },
+                axes: {
+                    "0": "scrollHorizontally",
+                    "1": "scrollVertically",
+                    "2": "vacant",
+                    "3": "vacant"
+                }
+            }
+        },
+        modelListeners: {
+            "axes.*": {
+                funcName: "{that}.produceNavigation",
+                args: "{change}"
+            },
+            "buttons.*": {
+                funcName: "{that}.produceNavigation",
+                args: "{change}"
+            }
+        },
+        intervalRecord: {
+            upwardScroll: null,
+            downwardScroll: null,
+            leftScroll: null,
+            rightScroll: null
+        },
+        invokers: {
+            produceNavigation: {
+                funcName: "gamepad.inputMapper.produceNavigation",
+                args: ["{that}", "{arguments}.0"]
+            },
+            scrollHorizontally: {
```

Comment:
I see now, you have separate methods for button and analogue inputs.  That makes sense.

Reply:
Yes, basically, we have `scrollLeft`, `scrollRight`, `scrollUp`, and `scrollDown` for buttons and triggers, while at the same time, `scrollHorizontally` and `scrollVertically` for axes (which use the former functions).

Reply:
I've written these in such a manner, keeping in mind that the user could reconfigure the controls at a later point in time.